apps/src/templates/instructions/codeReview/CommentOptions.jsx

```diff
@@ -13,13 +13,13 @@ class CommentOptions extends Component {
   };
 
   commentOptionTypes = {
-    markResolved: {
+    resolve: {
       onClick: this.props.onResolveStateToggle,
       iconClass: 'fa fa-fw fa-check',
       text: javalabMsg.resolve(),
       key: 'Resolve'
     },
```

Comment:
(previous comment didn't make it, so apologies for the double comment if it shows up again) Unlikely we would hit this case, but since we have explicit types for resolve/reopen here but are using the same callback (`onResolveStateToggle`) for each, could we get into a situation where the state is out of sync? Theoretically we shouldn't because it's still driven by `props.isResolved` here and `comment.isResolved` in ReviewTab but looking at just this component in isolation, that isn't immediately clear. I wonder then if it may be safer/clearer to have explicit `markResolved`/`markUnresolved` callbacks here? Again, I don't think it should but if in theory this component is used outside of the context of Code Review, it could have some issues related to that.

Reply:
mmm can you elaborate on how having two different callbacks would solve the potential issue? React semi-novice here...I think the `onResolveStateToggle` makes it pretty clear that we use the same callback to go both directions on whether a comment is resolved (this naming was something Molly suggested in a previous review, I called it originally onResolve which was a bad name).

Reply:
Sure, so yeah `onResolveStateToggle` does make the function clear, but my concern is just that if hypothetically this component is used outside the context of Code Review, a parent component could pass in an `onResolveStateToggle` function that flips the resolved state to the opposite of what the resolve/reopen is meant to do. Or in other words, there's an implicit association between this component's `props.isResolved` and the state of resolved inside of `onResolveStateToggle` that maybe isn't clear on first glance. Again I don't this is a likely case and as it's used now we shouldn't hit it, but I brought it up since the actual toggling happens in a different component and kind of implies it's tied to a single click/action? vs here there are two explicit actions (resolve/reopen) that even though effectively work as a toggle are defined differently. I'm fine with this as it just, but was just curious if it was worth exploring - basically it would involve changing the pattern to having some sort of `setResolveState(boolean)` type function in ReviewTab, and then here passing two functions as props corresponding to `markResolved` and `markReopened` or something, which could just be defined as
```
<CommentOptions
...
markResovled={() => {
   setResolveState(true);
}}
markReopened={() => {
  setResolveState(false);
}}
.../>
```
Totally up to you if you think it would make things clearer/cleaner since it would change up the callback logic, it works fine as it is now too.